packages/entrykit/src/onboarding/deposit/ChainSelect.tsx

```diff
@@ -80,33 +97,42 @@ export function ChainSelect({ value, onChange }: Props) {
         <Select.Portal container={frame.contentDocument.body}>
-          {/* TODO: hardcoded width */}
-          <Select.Content position="popper" className="w-[352px] mt-1 animate-in fade-in slide-in-from-top-2">
+          <Select.Content
+            position="popper"
+            className="w-[352px] max-h-[230px] overflow-y-auto mt-1 animate-in fade-in slide-in-from-top-2"
+          >
             <Select.Viewport>
               <Select.Group
                 className={twMerge(
                   "flex flex-col border divide-y",
                   "bg-neutral-800 text-neutral-300 border-neutral-700 divide-neutral-700",
                 )}
               >
-                {sourceChains.map((chain) => (
-                  // TODO: figure out why up/down arrow jump to top/bottom rather than cycling through items
-                  <Select.Item
-                    key={chain.id}
-                    value={chain.id.toString()}
-                    className={twMerge(
-                      "group flex p-2.5 gap-2.5 items-center cursor-pointer outline-none",
-                      // TODO: different style for checked/active state
-                      "text-white focus:bg-neutral-700 data-[state=checked]:bg-neutral-900",
-                    )}
-                  >
-                    <ChainIcon id={chain.id} name={chain.name} url={chain.relayChain?.icon?.[theme]} />
-                    <span className="flex-grow flex-shrink-0">{chain.name}</span>
-                    <span className="flex-shrink-0 font-mono text-sm text-neutral-400">
-                      <ChainBalance chainId={chain.id} address={userAccount.address} />
-                    </span>
-                  </Select.Item>
-                ))}
+                {isLoading ? (
+                  <div className="flex items-center justify-center p-4">
+                    <PendingIcon className="h-6 w-6 animate-spin text-gray-400" />
+                  </div>
+                ) : (
+                  [...(chainsWithBalance.length > 0 ? chainsWithBalance : sourceChains)].map((chain) => {
+                    const balance = chainsBalances?.find((cb) => cb.chain.id === chain.id)?.balance;
+                    return (
+                      <Select.Item
+                        key={chain.id}
+                        value={chain.id.toString()}
+                        className={twMerge(
+                          "group flex p-2.5 gap-2.5 items-center cursor-pointer outline-none",
+                          "text-white focus:bg-neutral-700 data-[state=checked]:bg-neutral-900",
+                        )}
+                      >
+                        <ChainIcon id={chain.id} name={chain.name} url={chain.relayChain?.icon?.[theme]} />
+                        <span className="flex-grow flex-shrink-0">{chain.name}</span>
+                        <span className="flex-shrink-0 font-mono text-sm text-neutral-400">
+                          <Balance wei={balance?.value ?? 0n} />
```

Comment:
maybe we shouldn't assume balance zero if we can't or haven't loaded it? I think it makes the UI unpolished if things from zero to something rather than blank or spinner to something.